In [2]:
%%sql
INSERT INTO MOVIES (MTITLE,RATING)
    SELECT DISTINCT MTITLE,RATING
    FROM DATASET;

 * sqlite:///Movies.db
23 rows affected.


[]

In [21]:
%%sql
INSERT INTO THEATERS (TNAME,LOCATION,PHONE)
    SELECT DISTINCT TNAME,LOCATION,PHONE
    FROM DATASET;

 * sqlite:///Movies.db
9 rows affected.


[]

In [10]:
%%sql
INSERT INTO ARTISTS (ANAME)
    SELECT DISTINCT CNAME
    FROM DATASET;

 * sqlite:///Movies.db
153 rows affected.


[]

In [15]:
%%sql
INSERT INTO CREDITS (MID,AID,CCODE)
SELECT DISTINCT MID, AID, CCODE 
FROM DATASET 
    JOIN MOVIES ON (MOVIES.MTITLE=DATASET.MTITLE)
    JOIN ARTISTS ON (ARTISTS.ANAME=DATASET.CNAME);

 * sqlite:///Movies.db
162 rows affected.


[]

In [23]:
%%sql
INSERT INTO SHOWS (MID,TID,STIME)
SELECT DISTINCT MID,TID,ShowTime
FROM DATASET
    JOIN MOVIES ON (DATASET.MTITLE=MOVIES.MTITLE)
    JOIN THEATERS ON (DATASET.TNAME=THEATERS.TNAME);

 * sqlite:///Movies.db
131 rows affected.


[]

3. **Check your work.**  
There are 9 theaters, 23 movies, 152 artists, 131 shows, and 161 credits.

Data Lines Up With Counts Above

4. **Now check the source data.**  
There are actually 13 duplicate rows in the original source data! Can you spot them? Write a SQL query that counts how many times each (MTitle, TName, Showtime, CCode, and CName) appears. Which ones appear more than once?

In [24]:
%%sql
Select count(MTITLE)
FROM DATASET

 * sqlite:///Movies.db
Done.


[(913,)]

In [25]:
%%sql
Select count(TNAME)
FROM DATASET

 * sqlite:///Movies.db
Done.


[(913,)]

In [27]:
%%sql
Select count(SHOWTIME)
FROM DATASET

 * sqlite:///Movies.db
Done.


[(913,)]

In [28]:
%%sql
Select count(CCODE)
FROM DATASET

 * sqlite:///Movies.db
Done.


[(913,)]

In [29]:
%%sql
Select count(CNAME)
FROM DATASET

 * sqlite:///Movies.db
Done.


[(913,)]

5. **Drop the `DATASET` table.**  
We don't need it anymore. 

In [ ]:
%%sql 
drop DATASET

6. **Save, commit, and push your work to GitHub.**  
Use the commit summary comment "Completed Part 3".